x-pack/test/functional/page_objects/gis_page.js

```diff
@@ -19,6 +19,14 @@ export function GisPageProvider({ getService, getPageObjects }) {
 
   class GisPage {
 
+    constructor() {
+      this.basePath = '';
+    }
+
+    setBasePath(basePath) {
+      this.basePath = basePath;
```

Comment:
This seems rather risky, as all other page objects are state-less... What's the lifetime of the `GisPage`?

Reply:
Yeah, I'm not sure what the lifetime of page objects are within the context of the FTR. I had the same gut-reaction too, but it's not a whole lot different from what we do with ES today when running these tests with before/after hooks to load/unload data.

We basically need a way to get the base path down into the internals of this page object, and this seemed less invasive than adding an argument to every function. We can alternatively disable the suite and have the GIS team decide how they want to handle it.

Reply:
How about we add an option to `PageObjects.common.navigateToUrl()` that maintains the current space, which would determine the space info using `browser.getUrl()` rather than using a generic `basePath`?

Reply:
Talked with @spalger, and we agree that a more holistic approach is needed for space-aware functional tests. We don't want to address that here/now given the sprawling nature of our `basePath: /s/custom_space` usage today.